### PR TITLE
completed css-two-column-layout challenge.

### PR DESCRIPTION
**Pull request by david-s-song**
css-two-column-layout

### CODE REVIEW

**Review by codingcodymiller — CHANGES_REQUESTED** [review]
Hey David! This is looking pretty good, but please revise this to make proper use of layout classes, such as `container`, `row`, and `column` classes. This works, but you should really be practicing setting up layout classes for yourself, they will be crucial as you start to deal with larger CSS tasks in projects.

**Comment by david-s-song** [comment]
css two column layout redo

File: css-two-column-layout/styles.css [hunk 1]
```diff
@@ -35,22 +36,25 @@ h4 {
...
   justify-content: space-evenly;
-  margin: 45px auto 0;
+  margin-top: 30px;
+  margin-bottom: 0;
+  margin-left: auto;
+  margin-right: auto;
 }
 
-.card {
+.column-one-fourth {
```
Comment (codingcodymiller) [inline css-two-column-layout/styles.css:57]: This class name isn't accurate. A fourth would be `25%`. You should name this something like `column-40`. **However**, I think this is actually something that should be a `column-half`. I suggest you revise this to `width: 50%` and make adjustments elsewhere to make that measurement correct, you may need to add `padding` to these elements to constrict their content a bit.

**Comment by david-s-song** [comment]
css two column redo

File: css-two-column-layout/styles.css [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+.row {
+  display: flex;
+  width: 90%;
+  flex-wrap: wrap;
+  justify-content: space-evenly;
+  margin-top: 30px;
+  margin-bottom: 0;
+  margin-left: auto;
+  margin-right: auto;
```
Comment (codingcodymiller) [inline css-two-column-layout/styles.css:54]: image

File: css-two-column-layout/styles.css [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+.column-half {
+  display: flex;
+  width: 50%;
+  margin-bottom: 25px;
+  padding-right: 5%;
+  padding-left: 5%;
```
Comment (codingcodymiller) [inline css-two-column-layout/styles.css:62]: image

File: css-two-column-layout/styles.css [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  flex-wrap: wrap;
+  justify-content: space-evenly;
+  margin-top: 30px;
+  margin-bottom: 0;
+  margin-left: auto;
+  margin-right: auto;
+}
+
+.column-half {
+  display: flex;
```
Comment (codingcodymiller) [inline css-two-column-layout/styles.css:58]: If you are applying `display: flex` to a column class, that usually means that the `column` should contain a `row` class element. That is the purpose of the `row` class, to make flex containers

File: css-two-column-layout/styles.css [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  padding-left: 5%;
+}
+
+img {
+  width: 100%;
+}
+
+.image {
+  flex-grow: 0;
+  flex-shrink: 0;
```
Comment (codingcodymiller) [inline css-two-column-layout/styles.css:71]: Why?

**Comment by david-s-song** [comment]
css-two-column-redo

File: css-two-column-layout/styles.css [hunk 6]
```diff
@@ -39,60 +39,53 @@ h4 {
...
   width: 50%;
-  margin-bottom: 25px;
-  padding-right: 5%;
-  padding-left: 5%;
+  display: flex;
+  padding-top: 20px;
+  padding-bottom: 20px;
+  padding-right: 7%;
+  padding-left: 7%;
+  position: relative;
```
Comment (codingcodymiller) [inline css-two-column-layout/styles.css:58]: This is still a bunch of excess styling on your `column` class. COlumn classes should only include a `width` or `flex-basis`. You should make an additional class you can add along with the `column-half` class.

**Pull request merged**